### DISCUSSION TITLE
"Leveling before print" feature from orca-flashforge. (adventurer 5m)

### DISCUSSION THREAD

**Discussion by mstraciak** [OrcaSlicer/OrcaSlicer#6018] [Ideas]
my question is focused on the flashforge adventurer 5m printer.
would it be possible to add the "leveling before print" function from orca-flashforge to orcaslicer? possibly also other control functions for this printer. orca-flashforge is not for linux and is therefore unavailable for some users.

**Comment by timnolte** [comment]
@mstraciak something I'm intrigued about, that someone in one of the Facebook groups mentioned, is if there is a gcode that can be sent to initiate the auto bed-leveling at the start of a print. From my experience having all print jobs auto bed-level is best.

I wonder though if it would be best for there to be a way to have it only add the gcode if the print after send option is used when sending over WiFi.

**Comment by DrJPK** [comment]
After slicing an stl using flashprint for the 5M Pro, and looking inside the resulting .gx file there is an unusual line in the start gcode

`Q603 MINX-12.50 MAXX109.20 MINY-110.00 MAXY110.00 Z4.40 T0`

this appears directly before the set bed temperature and set extruder temperature commands

I've not seen a Q code used before but it does appear to have boundaries for probing and a Z height to bounce to.  

My machine is out of commission right now with a broken filament feeder, but you could try adding this to the start g-code as part of a printer profile

**Reply by timnolte** [reply]
Ooh, I may give this a try on my 5M.

**Reply by timnolte** [reply]
So, I tried just adding this to the machine start gcode and then I selected upload and print from OrcaSlicer and it didn't auto-level the bed. :-/

**Comment by timnolte** [comment]
FYI, for anyone that wants a working bed leveling with gcode that also only does a bed mesh calibration for the area being printed on, with an allowance for cool build plates I have a gist here that you can use.

https://gist.github.com/timnolte/0e5e1b3780720baa8c6e64bacf4c5606
